src/code/middleware/firebase-decorator.ts

```diff
@@ -15,11 +15,11 @@ export const Firebasify = (model:any, relativeDataPath:string, callBack?:()=> vo
       updateFunc(newV);
       ref.on('value',updateFunc);
       onSnapshot(model, (newSnapshot:any) => {
-        const data = _.clone(newSnapshot);
+        let data = _.clone(newSnapshot);
```

Comment:
filterOutboundData() is already cloning the snapshot so it seems we can avoid the additional clone. What about something like
```
const data = model.filterOutboundData ? model.filterOutboundData(newSnapshot) : newSnapshot;
ref.update(data);
```

Reply:
Yes, much nicer.  Thanks!